loki/dimension.py

```diff
@@ -69,68 +81,90 @@ def variables(self):
         """
-        return f'{self._bounds[0]}:{self._bounds[1]}'
+        return self.indices[0] if self.indices else None
 
     @property
-    def size_expressions(self):
+    def lower(self):
         """
-        A list of all expression strings representing the size of a data space.
+        String or tuple of strings that matches the lower bound of the iteration space.
+        """
+        return self._lower[0] if self._lower and len(self._lower) == 1 else self._lower
 
-        This includes generic aliases, like ``end - start + 1`` or ``1:size`` ranges.
+    @property
+    def upper(self):
         """
-        exprs = as_tuple(self.size)
-        if self._aliases:
-            exprs += as_tuple(self._aliases)
-        exprs += (f'1:{self.size}', )
-        if self._bounds:
-            exprs += (f'{self._bounds[1]} - {self._bounds[0]} + 1', )
-        return exprs
+        String or tuple of strings that matches the upper bound of the iteration space.
+        """
+        return self._upper[0] if self._upper and len(self._upper) == 1 else self._upper
 
     @property
-    def bounds_expressions(self):
+    def step(self):
+        """
+        String or tuple of strings that matches the step size of the iteration space.
         """
-        A list of all expression strings representing the bounds of a data space.
+        return self._step[0] if self._step and len(self._step) == 1 else self._step
+
+    @property
+    def bounds(self):
         """
+        Tuple of expression string that represent the bounds of an iteration space.
 
-        exprs = [(b,) for b in self.bounds]
-        if self._bounds_aliases:
-            exprs = [expr + (b,) for expr, b in zip(exprs, self._bounds_aliases)]
+        .. note:
 
-        return as_tuple(exprs)
+        If mutiple lower or upper bound string have been provided,
+        only the first pair will be used.
+        """
+        return (
+            self.lower[0] if isinstance(self.lower, tuple) else self.lower,
+            self.upper[0] if isinstance(self.upper, tuple) else self.upper
+        )
 
     @property
-    def index_expressions(self):
+    def range(self):
         """
-        A list of all expression strings representing the index expression of an iteration space (loop).
+        String that matches the range expression of an iteration space (loop).
+
+        .. note:
+
+        If mutiple lower or upper bound string have been provided,
+        only the first pair will be used.
         """
+        return f'{self.bounds[0]}:{self.bounds[1]}'
 
-        exprs = [self.index,]
-        if self._index_aliases:
-            exprs += list(self._index_aliases)
+    @property
+    def size_expressions(self):
+        """
+        A list of all expression strings representing the size of a data space.
 
-        return as_tuple(exprs)
+        This includes generic aliases, like ``end - start + 1`` or ``1:size`` ranges.
+        """
+        exprs = as_tuple(self.size)
```

Comment:
As mentioned above: It might make sense to use `self.sizes` here to capture the aliases?